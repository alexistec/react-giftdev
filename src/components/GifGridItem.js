import React from 'react';
import propTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__bounce">
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title:propTypes.string.isRequired,
    url:propTypes.string.isRequired
}