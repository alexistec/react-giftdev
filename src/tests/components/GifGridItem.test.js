import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Test in <GifGridItem/>', () => {
    const url='http://localhost/static/img/k.jpg';
    const title = 'Tilte test';
    const wrapper = shallow( <GifGridItem url={ url } title={ title } /> );

    test('should render the component correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('should have a paragraph with a text',()=>{
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('we check the props of the img tag',()=>{
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('should have className animate__bounce', () => {
        const div = wrapper.find('div');  
        const className = div.prop('className');   
        expect( className.includes('animate__bounce') ).toBe( true )
    })
    
    
})
