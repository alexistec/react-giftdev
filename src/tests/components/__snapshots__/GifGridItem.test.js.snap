// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem/> should render the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Tilte test"
    src="http://localhost/static/img/k.jpg"
  />
  <p>
    Tilte test
  </p>
</div>
`;
